### PR TITLE
Add grpc example

### PR DESCRIPTION
**Pull request by vepatel**
### Proposed changes

- Add virtual-server grpc example

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that all unit tests pass after adding my changes
- [x] I have updated necessary documentation
- [x] I have rebased my branch onto main
- [x] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

### CODE REVIEW

**Review by jjngx — APPROVED** [review]

File: examples/custom-resources/grpc-upstreams/README.md [hunk 1]
```diff
@@ -5,32 +5,65 @@ grpc** field to an upstream. The protocol defaults to http if left unset.
...
-    type: grpc
-  routes:
-  - path: /helloworld.Greeter
-    action:
-      pass: grpc1
+2. VirtualServer and VirtualServerRoute resources for gRPC applications must include TLS termination.
+
+3. `grpcurl` utility must be installed
+
+4. Follow the [installation](https://docs.nginx.com/nginx-ingress-controller/installation/installation-with-manifests/)
```
Comment (ADubhlaoich) [inline examples/custom-resources/grpc-upstreams/README.md:15]:
```suggestion
4. [Install NGINX Ingress Controller using Manifests](https://docs.nginx.com/nginx-ingress-controller/installation/installation-with-manifests/)
```

File: examples/custom-resources/grpc-upstreams/README.md [hunk 2]
```diff
@@ -5,32 +5,65 @@ grpc** field to an upstream. The protocol defaults to http if left unset.
...
+
+5. Save the public IP address of the Ingress Controller into a shell variable:
```
Comment (ADubhlaoich) [inline examples/custom-resources/grpc-upstreams/README.md:17]:
```suggestion
5. Save the public IP address of NGINX Ingress Controller into a shell variable:
```

File: examples/custom-resources/grpc-upstreams/README.md [hunk 3]
```diff
@@ -5,32 +5,65 @@ grpc** field to an upstream. The protocol defaults to http if left unset.
...
+
+    ```shell
+    IC_IP=XXX.YYY.ZZZ.III
+    ```
+
+6. Save the HTTPS port of the Ingress Controller into a shell variable:
```
Comment (ADubhlaoich) [inline examples/custom-resources/grpc-upstreams/README.md:23]:
```suggestion
6. Save the HTTPS port of NGINX Ingress Controller into a shell variable:
```

File: examples/custom-resources/grpc-upstreams/README.md [hunk 4]
```diff
@@ -5,32 +5,65 @@ grpc** field to an upstream. The protocol defaults to http if left unset.
...
-  upstreams:
-  - name: grpc1
-    service: grpc-svc
-    port: 50051
-    type: grpc
-  routes:
-  - path: /helloworld.Greeter
-    action:
-      pass: grpc1
+2. VirtualServer and VirtualServerRoute resources for gRPC applications must include TLS termination.
```
Comment (ADubhlaoich) [inline examples/custom-resources/grpc-upstreams/README.md:11]:
```suggestion
2. Configure TLS termination for VirtualServer and VirtualServerRoute resources.
```

It's a requirement and in the context of gRPC applications, so mentioning it is redundant.

File: examples/custom-resources/grpc-upstreams/README.md [hunk 5]
```diff
@@ -5,32 +5,65 @@ grpc** field to an upstream. The protocol defaults to http if left unset.
 
 ## Prerequisites
 
-- HTTP/2 must be enabled. See `http2` ConfigMap key in the
+1. HTTP/2 must be enabled. See `http2` ConfigMap key in the
   [ConfigMap](https://docs.nginx.com/nginx-ingress-controller/configuration/global-configuration/configmap-resource/#listeners)
```
Comment (ADubhlaoich) [inline examples/custom-resources/grpc-upstreams/README.md:9]:
```suggestion
1. HTTP/2 must be enabled using the `http2` [ConfigMap key](https://docs.nginx.com/nginx-ingress-controller/configuration/global-configuration/configmap-resource/#listeners).
```

File: examples/custom-resources/grpc-upstreams/README.md [hunk 6]
```diff
@@ -5,32 +5,65 @@ grpc** field to an upstream. The protocol defaults to http if left unset.
...
-    service: grpc-svc
-    port: 50051
-    type: grpc
-  routes:
-  - path: /helloworld.Greeter
-    action:
-      pass: grpc1
+2. VirtualServer and VirtualServerRoute resources for gRPC applications must include TLS termination.
+
+3. `grpcurl` utility must be installed
```
Comment (ADubhlaoich) [inline examples/custom-resources/grpc-upstreams/README.md:13]:
```suggestion
3. A working [`grpcurl`](https://github.com/fullstorydev/grpcurl) installation.
```

File: examples/custom-resources/grpc-upstreams/README.md [hunk 7]
```diff
@@ -5,32 +5,65 @@ grpc** field to an upstream. The protocol defaults to http if left unset.
...
+
+2. Create the VirtualServer resource:
+
+    ```shell
+    kubectl create -f greeter-virtual-server.yaml
+    ```
+
+## Step 4 - Test the Configuration
+
+Access the application using `grpcurl`. We'll use `-insecure` option to turn off certificate verification of our self-signed certificate.
```
Comment (ADubhlaoich) [inline examples/custom-resources/grpc-upstreams/README.md:59]:
```suggestion
Access the application using `grpcurl`. Use the `-insecure` flag to turn off certificate verification for the self-signed certificate.
```

**Review by ADubhlaoich — APPROVED** [review]
LGTM generally: edit suggestions for nouns, punctuation and tone.

**Pull request merged**